### PR TITLE
Fetch: test URL encoding

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

**Comment by annevk** [comment]
Note that only Firefox gets this correct, but I'd really like to avoid this sadness from spreading so I'm going to file bugs against all other browsers.

**Comment by annevk** [comment]
* https://bugs.webkit.org/show_bug.cgi?id=178008
* https://bugs.chromium.org/p/chromium/issues/detail?id=772390
* https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14112807/

File: fetch/api/request/url-encoding.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<script>
+const expectedURL = new URL("?%C3%9F", location.href).href;
+const expectedURL2 = new URL("?%EF%BF%BD", location.href).href;
+test(() => {
+  let r = new Request("?\u00DF");
+  assert_equals(r.url, expectedURL);
+
+  r = new Request("?\uD83D");
+  assert_equals(r.url, expectedURL2);
+}, "URL encoding and Headers");
```
Comment (jdm) [inline fetch/api/request/url-encoding.html]: What do headers have to do with this?

Comment (annevk) [reply fetch/api/request/url-encoding.html]: Nothing, I keep mixing up the Headers and Request object in my head for some reason.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19238)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/284224509)

**Review by rakuco — APPROVED** [review]
No idea if my approval has much value, but the tests make sense to me at least.

**Comment by annevk** [comment]
Thanks, good enough for me!

**Pull request merged**